dev-util/plan9port/plan9port-20200330.ebuild

```diff
@@ -0,0 +1,132 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit multiprocessing toolchain-funcs
+
+MY_HASH="4650064aa757c217fa72f8819a2cf67c689bcdef"
+MY_P="${PN}-${MY_HASH}"
+
+DESCRIPTION="Port of many Plan 9 programs and libraries"
+HOMEPAGE="https://9fans.github.io/plan9port/"
+SRC_URI="https://github.com/9fans/${PN}/archive/${MY_HASH}.tar.gz -> ${MY_P}.tar.gz"
+
+LICENSE="9base BSD-4 MIT LGPL-2.1 BigelowHolmes"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="X aqua truetype"
+REQUIRED_USE="?? ( X aqua )"
+
+DEPEND="
+	X? ( x11-apps/xauth )
+	truetype? (
+		media-libs/freetype
+		media-libs/fontconfig
+	)
+"
+RDEPEND="${DEPEND}"
+
+PATCHES=(
+	"${FILESDIR}/${PN}-noexecstack.patch"
+	"${FILESDIR}/${PN}-cflags.patch"
+	"${FILESDIR}/${PN}-builderr.patch"
+)
+
+S="${WORKDIR}/${MY_P}"
+
+PLAN9="/usr/lib/plan9"
+EPLAN9="${EPREFIX}${PLAN9}"
+QA_MULTILIB_PATHS="${PLAN9}/.*/.*"
+
+src_prepare() {
+	default
+
+	case "${CHOST}" in
+		*freebsd10.*)  # patch 9l to use -pthread also on FreeBSD-10
+			eapply "${FILESDIR}/${PN}-freebsd-10.patch" ;;
+		*apple*)
+			sed -i 's/--noexecstack/-noexecstack/' src/mkhdr ||
+				die "Failed to sed AFLAGS" ;;
+	esac
+
+	# don't hardcode /bin and /usr/bin in PATH
+	sed -i '/PATH/s,/bin:/usr/bin:,,' INSTALL || die "sed on INSTALL failed"
+
+	# don't hardcode /usr/{,local/}include and prefix /usr/include/*
+	local f
+	for f in src/cmd/fontsrv/freetyperules.sh INSTALL $(find -name makefile); do
+		sed -Ei -e 's,-I/usr(|/local)/include ,,g' \
+			-e "s,-I/usr(|/local)/include,-I${EPREFIX}/usr\1/include,g" "${f}" ||
+			die "sed on ${f} failed"
+	done
+
+	# Fix paths, done in place of ./INSTALL -c
+	einfo "Fixing hard-coded /usr/local/plan9 paths"
+	grep -Zlr /usr/local/plan9 | xargs -0 sed -i "s,/usr/local/plan9,${EPLAN9},g"
+}
+
+src_configure() {
+	local -a myconf
+
+	if use X; then
+		myconf+=( X11="${EPREFIX}/usr" WSYSTYPE=x11 )
+	elif use aqua; then
+		local wsystype="$(awk '{if ($1 > 10.5) print "osx-cocoa"; else print "osx"}' \
+			<<< "${MACOSX_DEPLOYMENT_TARGET}")"
+		myconf+=( WSYSTYPE="${wsystype}" )
+	else
+		myconf+=( WSYSTYPE=nowsys )
+	fi
+
+	if use truetype; then
+		myconf+=( FONTSRV=fontsrv )
+	else
+		myconf+=( FONTSRV= )
+	fi
+
+	printf '%s\n' "${myconf[@]}" >> LOCAL.config
+}
+
+src_compile() {
+	export NPROC="$(makeopts_jobs)"
+	export CC9="$(tc-getCC)"
+
+	# The INSTALL script builds mk then [re]builds everything using that
+	einfo "Compiling Plan 9 from User Space can take a very long time"
+	einfo "depending on the speed of your computer. Please be patient!"
+	./INSTALL -b || die "Please report bugs to bugs.gentoo.org, NOT Plan9Port."
+}
+
+src_install() {
+	dodir "${PLAN9}"
+
+	# P9P's man does not handle compression
+	docompress -x "${PLAN9}/man"
+
+	# do* plays with the executable bit, and we should not modify them
+	cp -a * "${ED}/${PLAN9}"
```

Comment:
This is starting to look like it belongs in /opt instead...

Reply:
I think so too. blueness was using /usr/lib/plan9 but the documentation in install.txt prefers /opt/plan9 anyway.